parser/parser.go

```diff
@@ -781,15 +781,29 @@ func (p *parser) readService(srv *ast.Service) *parseError {
 		if err := p.readToken("("); err != nil {
 			return err
 		}
-
 		tok = p.next()
 		if tok.err != nil {
 			return tok.err
 		}
-		mth.InTypeName = tok.value // TODO: validate
+		if tok.value == "stream" {
```

Comment:
What about something more like this?

mth.InTypeName = tok.value // TODO: validate
if tok.value == "stream" {
  // If the next token isn't ")", this is a stream.
  tok = p.next()
  if tok.err != nil { return tok.err }
  if tok.value != ")" {
    mth.InTypeName = tok.value
    mth.ClientStreaming = true
  } else {
    p.back()
  }
}